Import does not compile

Stack trace:
```
 ~/Code/DisputeGameSimulations │ on honest_agent !4 ?1 cargo run                        │ took 8s │ 
   Compiling durin-fault v0.0.1 (https://github.com/anton-rs/durin.git#a3189f69)
error[E0599]: no function or associated item named `encode_single` found for struct `alloy_sol_types::sol_data::Uint` in the current scope
  --> /.cargo/git/checkouts/durin-59d0f5ca5f552eb9/a3189f6/crates/fault/src/providers/alphabet.rs:41:60
   |
41 |         let mut prestate_hash = keccak256(<sol!(uint256)>::encode_single(&prestate));
   |                                                            ^^^^^^^^^^^^^ function or associated item not found in `Uint<256>`

error[E0599]: no function or associated item named `encode` found for tuple `(alloy_sol_types::sol_data::Uint<256>, alloy_sol_types::sol_data::Uint<256>)` in the current scope
  --> /.cargo/git/checkouts/durin-59d0f5ca5f552eb9/a3189f6/crates/fault/src/providers/alphabet.rs:61:67
   |
61 |         let mut state_hash = keccak256(AlphabetClaimConstruction::encode(&state_sol));
   |                                                                   ^^^^^^ function or associated item not found in `(Uint<256>, Uint<256>)`

For more information about this error, try `rustc --explain E0599`.
error: could not compile `durin-fault` (lib) due to 2 previous errors
```
looks like a dependency version thing. I'll check it out try to ship a fix PR today

--- a/crates/fault/Cargo.toml
+++ b/crates/fault/Cargo.toml
@@ -10,8 +10,8 @@ version.workspace = true
 durin-primitives = { path = "../primitives" }
 
 # External
-alloy-primitives = { git = "https://github.com/alloy-rs/core.git" }
-alloy-sol-types = { git = "https://github.com/alloy-rs/core.git" }
+alloy-primitives = { version = "0.4.2" }
+alloy-sol-types = { version = "0.4.2" }
 anyhow = "1.0.75"
 
 [dev-dependencies]

--- a/crates/fault/src/providers/alphabet.rs
+++ b/crates/fault/src/providers/alphabet.rs
@@ -38,7 +38,7 @@ impl TraceProvider<[u8; 1]> for AlphabetTraceProvider {
 
     fn absolute_prestate_hash(&self) -> Claim {
         let prestate = U256::from(self.absolute_prestate);
-        let mut prestate_hash = keccak256(<sol!(uint256)>::encode_single(&prestate));
+        let mut prestate_hash = keccak256(<sol!(uint256)>::abi_encode(&prestate));
         prestate_hash[0] = VMStatus::Unfinished as u8;
         prestate_hash
     }
@@ -58,7 +58,7 @@ impl TraceProvider<[u8; 1]> for AlphabetTraceProvider {
             U256::from(position.trace_index(self.max_depth)),
             U256::from(self.state_at(position)?[0]),
         );
-        let mut state_hash = keccak256(AlphabetClaimConstruction::encode(&state_sol));
+        let mut state_hash = keccak256(AlphabetClaimConstruction::abi_encode(&state_sol));
         state_hash[0] = VMStatus::Invalid as u8;
         Ok(state_hash)
     }
@@ -82,7 +82,7 @@ mod test {
         };
 
         let prestate_sol = U256::from(provider.absolute_prestate()[0]);
-        let prestate = <sol!(uint256)>::encode_single(&prestate_sol);
+        let prestate = <sol!(uint256)>::abi_encode(&prestate_sol);
         assert_eq!(
             hex!("0000000000000000000000000000000000000000000000000000000000000061"),
             prestate.as_slice()
@@ -108,7 +108,8 @@ mod test {
             let position = compute_gindex(provider.max_depth, i as u64);
 
             let expected_encoded = (U256::from(i), U256::from(expected));
-            let mut expected_hash = keccak256(AlphabetClaimConstruction::encode(&expected_encoded));
+            let mut expected_hash =
+                keccak256(AlphabetClaimConstruction::abi_encode(&expected_encoded));
             expected_hash[0] = VMStatus::Invalid as u8;
 
             assert_eq!(provider.state_at(position).unwrap()[0], expected);

--- a/crates/primitives/Cargo.toml
+++ b/crates/primitives/Cargo.toml
@@ -6,5 +6,5 @@ authors = ["clabby"]
 version.workspace = true
 
 [dependencies]
-alloy-primitives = { git = "https://github.com/alloy-rs/core.git" }
+alloy-primitives = { version = "0.4.2" }
 anyhow = "1.0.75"

